### PR TITLE
test(cal): add team calendar scheduling coverage

### PR DESCRIPTION
**Pull request by vttranlina**
ref https://github.com/linagora/esn-sabre/issues/394

### CODE REVIEW

File: src/test/java/com/linagora/dav/contracts/cal/TeamCalendarSchedulingContract.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                calDavClient.getCalendarEvent(bobMember, bobMemberDelegatedCalendar.eventHref(eventUid)), nonMember.email()))
+            .as("Team Calendar event should reflect nonMember PARTSTAT %s".formatted(partStatValue))
+            .isEqualTo(partStat));
+
+        // And aliceMember attendee copy also reflects nonMember participation status
+        awaitAtMost.untilAsserted(() -> assertThat(CalendarUtil.getAttendeePartStat(
+                calDavClient.getCalendarEvent(aliceMember, aliceMemberEventUri), nonMember.email()))
+            .isEqualTo(partStat));
+    }
+
```
Comment (chibenwa) [inline src/test/java/com/linagora/dav/contracts/cal/TeamCalendarSchedulingContract.java:259]: Add:
 - attendee personal setting should not propagate to team calendar
 - alarm should correctly replicate (each attendee with his own and only his own while keeping all alarms in the team calendar)

Probably more edge cases are around but that's the 2 obvious I see...

Comment (hungphan227) [reply src/test/java/com/linagora/dav/contracts/cal/TeamCalendarSchedulingContract.java:259]: added

**Review by hungphan227 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
I thought about more tests (ITIP)

GIVEN bob has no access to team calendar T1
WHEN he receives an email (REPLY eg) for T1
THEN the ITIP fails

(the TEAM_CALENDAR_ID stuff cannot be used to bypass ACL)

**Review by chibenwa — APPROVED** [review]
